js/lib/lib.go

```diff
@@ -37,11 +38,17 @@ var (
 
 func GetCoreJS() *goja.Program {
 	once.Do(func() {
-		coreJs = goja.MustCompile(
+		program, err := goja.Parse(
 			"core-js/shim.min.js",
 			rice.MustFindBox("core-js").MustString("shim.min.js"),
-			true,
-		)
+			parser.WithDisableSourceMaps)
+		if err != nil {
+			panic(err)
+		}
+		coreJs, err = goja.CompileAST(program, false)
+		if err != nil {
+			panic(err)
+		}
```

Comment:
Why was this done? To disable sourcemap support and improve performance?

In any case it should be in a separate commit from the goja update that explains it.

Reply:
Unfortunately goja fixing sourcemaps support (which was apperantly a little broken) made it so now if there are sourcemap mappings it errors out. Given that we don't support sourcemaps I decided that it is best to disable it. 

Supporting them will need more work especially around archives and ... I hoped that given #1772  and the now initiative to just put the rest in goja, that I will reduce somewhat the amount of ... work needed.

tl;dr this is needed in order for the update to compile corejs at all and I disabled it elsewhere because it won't work with `k6 archive` but it also didn't work until now.

Reply:
I added an issue https://github.com/loadimpact/k6/issues/1789 as I forgot at the time

Reply:
OK, thanks for explaining. It still belongs in a separate commit with that explanation ;)

Reply:
I might not have stressed this enough - the tests don't pass if I don't change this as the core-js shim has sourceMap tag/line in it

Reply:
That's fine, I'm only arguing for including it as a separate commit on top the the goja update one, so we can document why it was done, instead of it being mixed with the goja changes which I usually gloss over.

Reply:
I can add it to the message,which I should've probably done :facepalm: , but I don't like it being in a separate commit

Reply:
:+1: for a commit message and a single commit, if the tests will be broken without the change. That said, are we sure that we didn't have source maps before? The line numbers of errors weren't that far off the mark, despite us having core.js and babel transformation... IIRC, they were only off by 1 because of https://github.com/loadimpact/k6/blob/ccffbcaaddd7f2d472cdced0da847fde2140ef88/js/initcontext.go#L198-L202

:confused:

Reply:
I am pretty sure because they were [disabled LONG time ago in babel](https://github.com/loadimpact/k6/commit/31819e6ffe966fc247f58d3e717e41037068747a) and the reason we are not seeing bigger differences is because [babel tries](https://github.com/loadimpact/k6/blob/ef717e1af2a92d35d39a3d815a69b6ab6b8c65e1/js/compiler/compiler.go#L45) :).

Reply:
it *tries* because it works terrible for the first 10-15 lines if I remember correctly

Reply:
ok then, :+1: for this and we can figure out proper source maps in the future in https://github.com/loadimpact/k6/issues/1789, after we get rid of core.js